src/notifications/events.rs

```diff
@@ -164,7 +158,37 @@ pub enum ClientboundNotification {
         id: String,
         role_id: String
     },
+    ServerUpdate {
+        id: String,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        nonce: Option<String>,
+        owner: String,
+        name: String,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        description: Option<String>,
+        channels: Vec<String>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        categories: Option<Vec<Category>>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        system_messages: Option<SystemMessageChannels>,
+
+        #[serde(default = "HashMap::new", skip_serializing_if = "HashMap::is_empty")]
+        roles: HashMap<String, Role>,
+        default_permissions: PermissionTuple,
 
+        #[serde(skip_serializing_if = "Option::is_none")]
+        icon: Option<File>,
+
+        #[serde(skip_serializing_if = "Option::is_none")]
+        banner: Option<File>,
+
+        #[serde(skip_serializing_if = "if_false", default)]
+        nsfw: bool
+    },
```

Comment:
This is a pretty major breaking API change and from the code earlier seems to needlessly resend the entire server object each time.